scripts/pipeline/orthology_benchmark.py

```diff
@@ -175,6 +177,64 @@ def get_core_names(species_names: List[str], host: str, port: int, user: str) ->
     return core_names
 
 
+def get_gtf_file(core_name: str, source_dir: str, target_dir: str) -> None:
+    """Finds and copies a GTF file for a specified core db.
+
+    Note:
+        The code utilises the structure of `production/ensemblftp` or MC's personal dir
+        to speed up file search.
+
+    Args:
+        core_name: Core db name.
+        source_dir: Path to the directory containing subdirectories with GTF files.
+            `production/ensemblftp` or MC's `nobackup/release_dumps`.
+        target_dir: Path to the directory where the GTF file will be copied.
+
+    Raises:
+        RuntimeError: If command to copy a file fails for any reason.
+
+    Warns:
+        UserWarning: If the GTF file was not found.
+
+    """
+    species_name = core_name.split("_core_")[0]
+    release = core_name.split("_core_")[1].split("_")[0]
+
+    parent_dir = os.path.join(source_dir, "release-" + release)
+    # gtf_dirs contains all subdirs which could contain the gtf file
+    # Vertebrates in `production/ensemblftp`:
+    gtf_dirs = [os.path.join(parent_dir, "gtf", species_name)]
+    # Vertebrates in MC's personal dir and
+    # all other divisions whether in `production/ensemblftp` or MC's dir
+    potential_divisions = [f.name for f in os.scandir(parent_dir) if f.is_dir()]
+    for division in potential_divisions:
```

Comment:
Since you're only iterating over `potential_divisions` once, why not access the `scandir` iterator directly in the for loop?

Reply:
Do you mean something like:
1) `for division in [f.name for f in os.scandir(parent_dir) if f.is_dir()]:` -> I think the current solution is more readable
2) `gtf_dirs += [os.path.join(parent_dir, f.name, "gtf", species_name) for f in os.scandir(parent_dir) if f.is_dir()]` -> Not sure I can read this without getting dizzy
3) Something else I can't think of at the moment?

Reply:
I was thinking more along the lines of:
```suggestion
    for f in os.scandir(parent_dir):
        if f.is_dir():
            gtf_dirs.append(os.path.join(parent_dir, f.name, "gtf", species_name))
```
In fairness, this is not very concise, but I'd venture that it's not entirely unreadable.

Reply:
Thank you very much @twalsh-ebi. I've incorporated this locally so I will mark it as resolved here. I might swap it with the suggestion from Slack, tho 😄  But let me make the first iteration incorporating all the feedback from here 😉